File: yuema-miniprogram/services/venueService.js
const request = require('../utils/request');

module.exports = {
  // 获取场地列表
  getVenueList() {
    return request.get('/venue/list');
  },

  // 获取附近场地
  getNearbyVenues(longitude, latitude, limit = 10) {
    return request.get('/venue/nearby', { longitude, latitude, limit });
  },

  // 获取场地详情
  getVenueDetail(venueId) {
    return request.get('/venue/detail', { venueId });
  },

  /** 与 getVenueDetail 同义，兼容旧页面调用 */
  getVenueInfo(venueId) {
    return request.get('/venue/detail', { venueId });
  },

  // 获取场地包间
  getVenueRooms(venueId) {
    return request.get('/venue/rooms', { venueId });
  }
};

File: yuema-server/src/main/java/com/yuema/server/service/VenueService.java
package com.yuema.server.service;

import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
import com.yuema.server.entity.Venue;
import com.yuema.server.entity.VenueRoom;
import com.yuema.server.mapper.VenueMapper;
import com.yuema.server.mapper.VenueRoomMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class VenueService extends ServiceImpl<VenueMapper, Venue> {

    @Autowired
    private VenueRoomMapper venueRoomMapper;

    public List<Venue> getActiveVenues() {
        return baseMapper.selectActiveVenues();
    }

    public List<Venue> getNearbyVenues(Double longitude, Double latitude, Integer limit) {
        if (limit == null || limit <= 0) {
            limit = 10;
        }
        return baseMapper.selectNearbyVenues(longitude, latitude, limit);
    }

    public List<VenueRoom> getVenueRooms(Long venueId) {
        return venueRoomMapper.selectByVenueId(venueId);
    }

    public VenueRoom getRoomDetail(Long roomId) {
        return venueRoomMapper.selectById(roomId);
    }
}
